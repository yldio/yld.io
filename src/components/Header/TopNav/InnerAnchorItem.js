import React from 'react'
import styled from 'styled-components'

import outlineStyles from '../outlineStyles'
import Anchor from '../../Common/Anchor'
import headerItemStyles from '../headerItemStyles'
import topNavItemStyles from './topNavItemStyles'

const InnerListItem = styled.li`
  display: flex;
`

const InnerAnchor = styled(Anchor)`
  ${headerItemStyles}
  ${topNavItemStyles}
  ${outlineStyles}

  width: 100%;
  background: ${props => props.theme.colors.greyBG};
  color: ${props => props.theme.colors.textLight};

  &:hover,
  &.active {
    color: ${props => props.theme.colors.text};
  }
`

export const InnerAnchorItem = ({
  children,
  to,
  href,
  activeClassName,
  themeVariation,
  ...props
}) => (
  <InnerListItem themeVariation={themeVariation} {...props}>
    <InnerAnchor href={href} to={to} activeClassName={activeClassName}>
      {children}
    </InnerAnchor>
  </InnerListItem>
)

export default InnerAnchorItem
