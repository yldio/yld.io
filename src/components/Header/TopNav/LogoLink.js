import React, { Fragment, useState } from 'react'
import { Link } from 'gatsby'
import remcalc from 'remcalc'
import styled from 'styled-components'

import logo from '../../../images/logo_animated.gif'
import ServiceSpecialityLogo from '../../../images/service-speciality-logo'
import {
  servicesRegExp,
  logoColors,
  getSpeciality,
  getService
} from '../navLinksHelper'

const StyledLink = styled(Link)`
  height: ${remcalc(48)};
  width: ${remcalc(48)};

  @media screen and (min-width: 960px) {
    height: ${remcalc(54)};
    width: ${remcalc(54)};
  }
`

const LogoLink = ({ path = '/' }) => {
  const isServicePage = path.search(servicesRegExp) > -1
  const isSpecialityPage = path.includes('speciality')

  const serviceTitle = isServicePage ? path.match(servicesRegExp)[0] : null
  const service = isSpecialityPage ? getService(path) : serviceTitle
  const speciality = isSpecialityPage ? getSpeciality(path) : null

  let originalFillColor = isSpecialityPage
    ? logoColors[service][speciality]
    : logoColors['default']
  const [fillColor, setFillColor] = useState(originalFillColor)

  return (
    <Fragment>
      {isSpecialityPage || isServicePage ? (
        <StyledLink
          to="/"
          onMouseEnter={() =>
            setFillColor(
              logoColors[isServicePage ? 'defaultHover' : 'specialityHover']
            )
          }
          onMouseLeave={() => setFillColor(originalFillColor)}
        >
          <ServiceSpecialityLogo
            fillColor={fillColor}
            textColor={
              logoColors[isServicePage ? 'defaultText' : 'specialityText']
            }
          />
        </StyledLink>
      ) : (
        <Link to="/">
          <img role="link" height="48" src={logo} alt="yld logo" />
        </Link>
      )}
    </Fragment>
  )
}

export default LogoLink
