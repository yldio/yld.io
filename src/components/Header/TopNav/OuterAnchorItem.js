import React from 'react'
import styled from 'styled-components'

import Anchor from '../../Common/Anchor'
import headerItemStyles from '../headerItemStyles'
import outlineStyles from '../outlineStyles'
import topNavItemStyles from './topNavItemStyles'
import TopNavItem from './TopNavItem'

const StyledAnchor = styled(Anchor)`
  ${headerItemStyles}
  ${topNavItemStyles}

  ${outlineStyles}
`

const StyledListItem = styled(TopNavItem)`
  display: flex;
  ${props => props.states.default}

  &:hover,
  > a:hover {
    ${props => props.states.hover}
  }

  > a:active {
    ${props => props.states.clickTap}
  }

  > a.active {
    ${props => props.states.active}
    &:active {
      ${props => props.states.active}
    }

    &:hover {
      ${props => props.states.activeAndHover}
    }
  }
`

const OuterAnchorItem = ({
  to,
  href,
  activeClassName,
  themeVariation,
  onClick,
  label,
  ...props
}) => (
  <StyledListItem themeVariation={themeVariation} {...props}>
    <StyledAnchor
      href={href}
      to={to}
      activeClassName={activeClassName}
      onClick={onClick}
      title={label}
    >
      {label}
    </StyledAnchor>
  </StyledListItem>
)

export default OuterAnchorItem
