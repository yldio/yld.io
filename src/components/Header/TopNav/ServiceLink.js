import React, { Fragment } from 'react'
import { Link } from 'gatsby'
import styled from 'styled-components'
import remcalc from 'remcalc'
import { capitalize } from 'lodash'

import { servicesRegExp, getIsServicePage, getService } from '../navLinksHelper'

const StyledServiceLink = styled(Link)`
  font-size: ${remcalc(26)};
  margin-left: ${remcalc(12)};
  color: ${props => props.theme.colors[props.color]};

  @media screen and (min-width: 960px) {
    font-size: ${remcalc(30)};
  }

  &:hover {
    text-decoration: underline;
    color: ${props => props.theme.colors[props.color]};
  }
`

const ServiceLink = ({ path = '/' }) => {
  const isServicePage = getIsServicePage(path)
  const serviceTitle = isServicePage ? path.match(servicesRegExp)[0] : null

  const isSpecialityPage = path.includes('speciality')
  const service = isSpecialityPage ? getService(path) : serviceTitle

  return (
    <Fragment>
      {isServicePage || isSpecialityPage ? (
        <StyledServiceLink
          to={`/${service}`}
          color={isServicePage ? 'text' : 'white'}
        >
          {capitalize(service)}
        </StyledServiceLink>
      ) : null}
    </Fragment>
  )
}

export default ServiceLink
