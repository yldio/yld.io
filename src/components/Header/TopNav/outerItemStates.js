import { css } from 'styled-components'

const defaultLight = css`
  background: ${props => props.theme.colors.white};
  color: ${props => props.theme.colors.text};
  opacity: 1;
`

const hoverLight = css`
  background: ${props => props.theme.colors.greyBG};
  color: ${props => props.theme.colors.text};
  opacity: 1;
`

const clickTapLight = css`
  background: ${props => props.theme.colors.text};
  color: ${props => props.theme.colors.white};
  opacity: 1;
`

const activeLight = css`
  background: ${props => props.theme.colors.white};
  color: ${props => props.theme.colors.text};
  opacity: 0.5;
`

const activeAndHoverLight = css`
  background: ${props => props.theme.colors.text};
  color: #a9a9a9;
  opacity: 1;
`

const defaultDark = css`
  background: ${props => props.theme.colors.blueBg};
  color: ${props => props.theme.colors.white};
  opacity: 1;
`

const hoverDark = css`
  background: #3a3553;
  color: ${props => props.theme.colors.white};
  opacity: 1;
`

const clickTapDark = css`
  background: ${props => props.theme.colors.vibrant};
  color: ${props => props.theme.colors.text};
  opacity: 1;
`

const activeDark = css`
  background: ${props => props.theme.colors.blueBg};
  color: ${props => props.theme.colors.white};
  opacity: 0.7;
`

const activeAndHoverDark = css`
  background: ${props => props.theme.colors.vibrant};
  color: #007f56;
  opacity: 1;
`

export const lightStates = {
  default: defaultLight,
  hover: hoverLight,
  clickTap: clickTapLight,
  active: activeLight,
  activeAndHover: activeAndHoverLight
}

export const darkStates = {
  default: defaultDark,
  hover: hoverDark,
  clickTap: clickTapDark,
  active: activeDark,
  activeAndHover: activeAndHoverDark
}
