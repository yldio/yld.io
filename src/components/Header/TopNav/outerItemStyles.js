import { css } from 'styled-components'

export const defaultLightStyles = css`
  background: ${props => props.theme.colors.white};
  color: ${props => props.theme.colors.text};
  opacity: 1;
`

export const hoverLightStyles = css`
  background: ${props => props.theme.colors.greyBG};
  color: ${props => props.theme.colors.text};
  opacity: 1;
`

export const clickTapLightStyles = css`
  background: ${props => props.theme.colors.text};
  color: ${props => props.theme.colors.white};
  opacity: 1;
`

export const activeLightStyles = css`
  background: ${props => props.theme.colors.white};
  color: ${props => props.theme.colors.text};
  opacity: 0.5;
`

export const activeAndHoverLightStyles = css`
  background: ${props => props.theme.colors.text};
  color: #a9a9a9;
  opacity: 1;
`

export const defaultDarkStyles = css`
  background: ${props => props.theme.colors.blueBg};
  color: ${props => props.theme.colors.white};
  opacity: 1;
`

export const hoverDarkStyles = css`
  background: #3a3553;
  color: ${props => props.theme.colors.white};
  opacity: 1;
`

export const clickTapDarkStyles = css`
  background: ${props => props.theme.colors.vibrant};
  color: ${props => props.theme.colors.text};
  opacity: 1;
`

export const activeDarkStyles = css`
  background: ${props => props.theme.colors.blueBg};
  color: ${props => props.theme.colors.white};
  opacity: 0.7;
`

export const activeAndHoverDarkStyles = css`
  background: ${props => props.theme.colors.vibrant};
  color: #007f56;
  opacity: 1;
`
