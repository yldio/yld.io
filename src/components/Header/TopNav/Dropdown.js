import React, { PureComponent } from 'react'
import styled, { css } from 'styled-components'
import is from 'styled-is'
import remcalc from 'remcalc'

import Chevron from '../../Common/Chevron'
import InnerAnchorItem from './InnerAnchorItem'
import headerItemStyles from '../headerItemStyles'
import outlineStyles from '../outlineStyles'
import topNavItemStyles from './topNavItemStyles'
import TopNavItem from './TopNavItem'

const DropdownContainer = styled(TopNavItem)`
  position: relative;
  cursor: pointer;
  background: transparent;

  > span {
    ${props => props.states.default}

    &:hover {
      ${props => props.states.hover}
    }
  }

  ${is('expanded')`
    > span {
      ${props => props.states.clickTap}

      &:hover {
        ${props => props.states.activeAndHover}
      }
    }
  `}
`

const DropdownNameWrapper = styled.span`
  display: flex;
  align-items: center;
  transition: outline ${props => props.theme.animations.normal} ease-out;
  /* bumping the z-index so that the outline doesn't get behind the dropdown items list */
  z-index: 2;
  ${headerItemStyles}
  ${topNavItemStyles}
  ${props => !props.clicked && outlineStyles}

  ${props =>
    props.clicked &&
    props.themeVariation === props.theme.variations.dark &&
    props.expanded === true &&
    css`
      &:focus {
        outline-color: ${props.theme.colors.white};
      }
    `}
`

const DropdownName = styled.span`
  padding-right: ${remcalc(6)};
  outline: none;
`

const DropdownList = styled.ul`
  position: absolute;
  width: ${remcalc(160)};
  display: flex;
  flex-direction: column;
  top: ${remcalc(48)};
  left: -9999px;
  opacity: 0;
  transition: opacity ${props => props.theme.animations.normal} ease;
  background: ${props => props.theme.colors.greyBG};
  z-index: ${props => props.theme.zIndexes.header};
  ${is('expanded')`
    left: 0;
    opacity: 1;
  `};
`

export default class Dropdown extends PureComponent {
  constructor(props) {
    super(props)
    this.state = {
      isExpanded: false,
      clicked: false
    }
  }

  /**
   * The outline is not desired when the user opens the dropdown with the mouse
   * so we're detecting it on mouse down
   */
  handleMouseDown = () => {
    this.setState({ clicked: true, isExpanded: !this.state.isExpanded })
  }

  handleFocus = () => {
    this.setState({ isExpanded: true })
  }

  handleBlur = () => {
    this.setState({ isExpanded: false, clicked: false })
  }

  render() {
    const { items, themeVariation, children } = this.props
    const { isExpanded, clicked } = this.state

    return (
      <DropdownContainer
        expanded={isExpanded}
        aria-haspopup="true"
        aria-expanded={isExpanded}
        onFocus={this.handleFocus}
        onMouseDown={this.handleMouseDown}
        onBlur={this.handleBlur}
        themeVariation={themeVariation}
      >
        <DropdownNameWrapper
          tabIndex="0"
          expanded={isExpanded}
          clicked={clicked}
          themeVariation={themeVariation}
        >
          <DropdownName>{children}</DropdownName>
          <Chevron direction={isExpanded ? 'up' : 'down'} />
        </DropdownNameWrapper>
        <DropdownList expanded={isExpanded}>
          {items.map(({ to, href, label }, idx) => (
            <InnerAnchorItem
              key={idx}
              themeVariation={themeVariation}
              href={href}
              to={to}
              activeClassName="active"
            >
              {label}
            </InnerAnchorItem>
          ))}
        </DropdownList>
      </DropdownContainer>
    )
  }
}
