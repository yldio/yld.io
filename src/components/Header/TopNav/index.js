import React, { Fragment } from 'react'
import styled from 'styled-components'
import breakpoint from 'styled-components-breakpoint'
import remcalc from 'remcalc'

import LogoLink from './LogoLink'
import ServiceLink from './ServiceLink'
import OuterAnchorItem from './OuterAnchorItem'
import Dropdown from './Dropdown'

const StyledLinksContainer = styled.div`
  display: flex;
  align-items: center;
`

const TopNavList = styled.ul`
  ${breakpoint('smallPhone')`
    display: none;
  `}

  @media screen and (min-width: 960px) {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    flex: 1;
    padding: ${remcalc(20)} ${remcalc(0)} ${remcalc(16)};
    padding-right: ${remcalc(0)};
  }
`

const TopNav = ({ links, themeVariation, path }) => (
  <Fragment>
    <StyledLinksContainer>
      <LogoLink path={path} />
      <ServiceLink path={path} />
    </StyledLinksContainer>
    <nav>
      <TopNavList>
        {links.map((link, idx) => {
          if (link.dropdownItems) {
            const { label, dropdownItems } = link
            return (
              <Dropdown
                key={idx}
                themeVariation={themeVariation}
                items={dropdownItems}
              >
                {label}
              </Dropdown>
            )
          } else {
            const { label, to, href } = link
            return (
              <OuterAnchorItem
                key={idx}
                themeVariation={themeVariation}
                activeClassName="active"
                to={to}
                href={href}
              >
                {label}
              </OuterAnchorItem>
            )
          }
        })}
      </TopNavList>
    </nav>
  </Fragment>
)

export default TopNav
