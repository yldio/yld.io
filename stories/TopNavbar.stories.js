import React from 'react'
import { storiesOf, addDecorator } from '@storybook/react'
import Theme from './theme'
import TopNav from '../src/components/Header/TopNav/'

addDecorator(Theme)

const anchors = [
  {
    label: 'Here',
    to: '/'
  },
  {
    label: 'There',
    to: '/there'
  },
  {
    label: 'LinkedIn (external anchor)',
    to: 'https://uk.linkedin.com/'
  }
]

const anchorsAndDropdowns = [
  {
    label: 'First dropdown',
    dropdownItems: [
      {
        label: 'Item 1',
        to: '/item-1/'
      },
      {
        label: 'Item 2',
        to: '/item-2/'
      },
      {
        label: 'Item 3',
        to: '/item-3/'
      }
    ]
  },
  {
    label: 'First anchor',
    to: '/anchor/'
  },
  {
    label: 'LinkedIn (external anchor)',
    href: 'https://uk.linkedin.com/'
  },
  {
    label: 'Second dropdown',
    dropdownItems: [
      {
        label: 'Go here',
        to: '/go-here/'
      },
      {
        label: 'Go there',
        to: '/go-there/'
      }
    ]
  },
  {
    label: 'Last anchor',
    to: '/last-anchor/'
  }
]

storiesOf('Header', module)
  .add('TopNavbar - light theme', () => (
    <TopNav links={anchors} themeVariation="light" />
  ))
  .add('TopNavbar with dropdowns - light theme', () => (
    <TopNav links={anchorsAndDropdowns} themeVariation="light" />
  ))
