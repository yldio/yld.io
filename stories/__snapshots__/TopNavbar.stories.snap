// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Header TopNavbar - light theme 1`] = `
.c4 {
  -webkit-transition: opacity 200ms ease-out, color 200ms ease-out;
  transition: opacity 200ms ease-out, color 200ms ease-out;
  outline: none;
  font-weight: 400;
  font-size: 1.0625rem;
  line-height: 1.5rem;
  padding: 0.625rem 0.9375rem 0.875rem;
  outline: 0.25rem solid transparent;
}

.c4:focus {
  outline-color: #65FFCD;
  z-index: 1;
}

.c4:active {
  outline: none;
}

.c3 {
  -webkit-tap-highlight-color: rgba(0,0,0,0);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background: #fff;
  color: #333333;
}

.c3:hover,
.c3 > a:hover {
  background: #333333;
  color: #a9a9a9;
}

.c3 > a:active {
  background: #333333;
  color: #fff;
}

.c3 > a.active {
  background: #fff;
  color: #828282;
}

.c3 > a.active:active {
  background: #fff;
  color: #828282;
}

.c3 > a.active:hover {
  color: #333333;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
  height: 5.25rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  display: none;
}

@media screen and (min-width:960px) {
  .c3 {
    margin-right: 0.375rem;
  }

  .c3:last-child {
    margin-right: 0rem;
  }
}

@media screen and (min-width:960px) {
  .c2 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
    padding: 1.25rem 0rem 1rem;
    padding-right: 0rem;
  }
}

<nav
  className="c0"
>
  <div
    className="c1"
  >
    <a
      aria-current="page"
      className=""
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      style={Object {}}
      title="Return to Homepage"
    >
      <img
        alt="yld logo"
        height="48"
        role="link"
        src="test-file-stub"
      />
    </a>
  </div>
  <ul
    className="c2"
  >
    <li
      className="c3"
    >
      <a
        aria-current="page"
        className="c4 active"
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
        style={Object {}}
        title="Here"
      >
        Here
      </a>
    </li>
    <li
      className="c3"
    >
      <a
        className="c4"
        href="/there"
        onClick={[Function]}
        onMouseEnter={[Function]}
        title="There"
      >
        There
      </a>
    </li>
    <li
      className="c3"
    >
      <a
        className="c4"
        href="/https:/uk.linkedin.com/"
        onClick={[Function]}
        onMouseEnter={[Function]}
        title="LinkedIn (external anchor)"
      >
        LinkedIn (external anchor)
      </a>
    </li>
  </ul>
</nav>
`;

exports[`Storyshots Header TopNavbar with dropdowns - light theme 1`] = `
.c11 {
  -webkit-transition: opacity 200ms ease-out, color 200ms ease-out;
  transition: opacity 200ms ease-out, color 200ms ease-out;
  outline: none;
  font-weight: 400;
  font-size: 1.0625rem;
  line-height: 1.5rem;
  padding: 0.625rem 0.9375rem 0.875rem;
  outline: 0.25rem solid transparent;
}

.c11:focus {
  outline-color: #65FFCD;
  z-index: 1;
}

.c11:active {
  outline: none;
}

.c10 {
  -webkit-tap-highlight-color: rgba(0,0,0,0);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background: #fff;
  color: #333333;
}

.c10:hover,
.c10 > a:hover {
  background: #333333;
  color: #a9a9a9;
}

.c10 > a:active {
  background: #333333;
  color: #fff;
}

.c10 > a.active {
  background: #fff;
  color: #828282;
}

.c10 > a.active:active {
  background: #fff;
  color: #828282;
}

.c10 > a.active:hover {
  color: #333333;
}

.c6 {
  stroke: currentColor;
  stroke-width: 0.1875rem;
  fill: none;
  -webkit-transform: rotate(135deg);
  -ms-transform: rotate(135deg);
  transform: rotate(135deg);
  -webkit-transition: -webkit-transform 0s ease 0.1s;
  -webkit-transition: transform 0s ease 0.1s;
  transition: transform 0s ease 0.1s;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c9 {
  -webkit-transition: opacity 200ms ease-out, color 200ms ease-out;
  transition: opacity 200ms ease-out, color 200ms ease-out;
  outline: none;
  font-weight: 400;
  font-size: 1.0625rem;
  line-height: 1.5rem;
  padding: 0.625rem 0.9375rem 0.875rem;
  outline: 0.25rem solid transparent;
  width: 100%;
  background: #f9f9f9;
  color: #828282;
}

.c9:focus {
  outline-color: #65FFCD;
  z-index: 1;
}

.c9:active {
  outline: none;
}

.c9:hover,
.c9.active {
  color: #333333;
}

.c3 {
  -webkit-tap-highlight-color: rgba(0,0,0,0);
  position: relative;
  cursor: pointer;
  background: transparent;
  -webkit-tap-highlight-color: rgba(0,0,0,0);
}

.c3 > span {
  background: #fff;
  color: #333333;
}

.c3 > span:hover {
  background: #333333;
  color: #a9a9a9;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: outline 300ms ease-out;
  transition: outline 300ms ease-out;
  z-index: 2;
  -webkit-transition: opacity 200ms ease-out, color 200ms ease-out;
  transition: opacity 200ms ease-out, color 200ms ease-out;
  outline: none;
  font-weight: 400;
  font-size: 1.0625rem;
  line-height: 1.5rem;
  padding: 0.625rem 0.9375rem 0.875rem;
  outline: 0.25rem solid transparent;
}

.c4:focus {
  outline-color: #65FFCD;
  z-index: 1;
}

.c4:active {
  outline: none;
}

.c5 {
  padding-right: 0.375rem;
  outline: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c7 {
  position: absolute;
  width: 10rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  top: 3rem;
  left: -9999px;
  opacity: 0;
  -webkit-transition: opacity 300ms ease;
  transition: opacity 300ms ease;
  background: #f9f9f9;
  z-index: 999;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
  height: 5.25rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  display: none;
}

@media screen and (min-width:960px) {
  .c10 {
    margin-right: 0.375rem;
  }

  .c10:last-child {
    margin-right: 0rem;
  }
}

@media screen and (min-width:960px) {
  .c3 {
    margin-right: 0.375rem;
  }

  .c3:last-child {
    margin-right: 0rem;
  }
}

@media screen and (min-width:960px) {
  .c2 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
    padding: 1.25rem 0rem 1rem;
    padding-right: 0rem;
  }
}

<nav
  className="c0"
>
  <div
    className="c1"
  >
    <a
      aria-current="page"
      className=""
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      style={Object {}}
      title="Return to Homepage"
    >
      <img
        alt="yld logo"
        height="48"
        role="link"
        src="test-file-stub"
      />
    </a>
  </div>
  <ul
    className="c2"
  >
    <li
      aria-expanded={false}
      aria-haspopup="true"
      className="c3"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
    >
      <span
        className="c4"
        tabIndex="0"
      >
        <span
          className="c5"
          data-event={null}
        >
          First dropdown
        </span>
        <svg
          className="c6"
          height="6"
          viewBox="0 0 6 6"
          width="6"
        >
          <polyline
            points="0,0 6,0 6,6"
          />
        </svg>
      </span>
      <ul
        className="c7"
      >
        <li
          className="c8"
        >
          <a
            className="c9"
            href="/item-1/"
            onClick={[Function]}
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            title="Item 1"
          >
            Item 1
          </a>
        </li>
        <li
          className="c8"
        >
          <a
            className="c9"
            href="/item-2/"
            onClick={[Function]}
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            title="Item 2"
          >
            Item 2
          </a>
        </li>
        <li
          className="c8"
        >
          <a
            className="c9"
            href="/item-3/"
            onClick={[Function]}
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            title="Item 3"
          >
            Item 3
          </a>
        </li>
      </ul>
    </li>
    <li
      className="c10"
    >
      <a
        className="c11"
        href="/anchor/"
        onClick={[Function]}
        onMouseEnter={[Function]}
        title="First anchor"
      >
        First anchor
      </a>
    </li>
    <li
      className="c10"
    >
      <a
        className="c11"
        href="https://uk.linkedin.com/"
        onClick={[Function]}
        rel="noopener noreferrer"
        target="_blank"
        title="LinkedIn (external anchor)"
      >
        LinkedIn (external anchor)
      </a>
    </li>
    <li
      aria-expanded={false}
      aria-haspopup="true"
      className="c3"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
    >
      <span
        className="c4"
        tabIndex="0"
      >
        <span
          className="c5"
          data-event={null}
        >
          Second dropdown
        </span>
        <svg
          className="c6"
          height="6"
          viewBox="0 0 6 6"
          width="6"
        >
          <polyline
            points="0,0 6,0 6,6"
          />
        </svg>
      </span>
      <ul
        className="c7"
      >
        <li
          className="c8"
        >
          <a
            className="c9"
            href="/go-here/"
            onClick={[Function]}
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            title="Go here"
          >
            Go here
          </a>
        </li>
        <li
          className="c8"
        >
          <a
            className="c9"
            href="/go-there/"
            onClick={[Function]}
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            title="Go there"
          >
            Go there
          </a>
        </li>
      </ul>
    </li>
    <li
      className="c10"
    >
      <a
        className="c11"
        href="/last-anchor/"
        onClick={[Function]}
        onMouseEnter={[Function]}
        title="Last anchor"
      >
        Last anchor
      </a>
    </li>
  </ul>
</nav>
`;
